enhancements/proposed/20191001-background-service-to-update-dns-resolvers-on-host-os-for-ingress-resources.md

```diff
@@ -0,0 +1,174 @@
+* Install local ingress ssl certificates on the Host OS with this service
+
+## Design Details
+
+### Design summary
+
+There are 3 aspects of making the ideal solution work seemlessly for the end user although #1 and #3 are not part 
+of this proposal
+
+#### 1. A DNS service that runs in the cluster and resolves ingress host names to the minikube cluster ip
+Use minikube as a DNS server for and have a DNS server which will resolve the DNS queries for ingress hosts to the IP 
+address of the minikube instance. This is the most preferable solution since the ingress resources within the cluster 
+would be limited to the cluster itself. For DNS resolution the A record for each DNS query would be the hostname of 
+the ingress and the ip address would be the cluster ip. There is a working solution for a plugin that currently does 
+this here: https://github.com/kubernetes/minikube/pull/5507
+
+#### 2. A background service that runs on the host OS that monitors changes to host names and minikube ips
+Run a background service that will update `/etc/resolver` or similar files on the Host OS as each new minikube instance
+is launched and stopped and as there are updates, removals, or additions to the ingress resources installed in any 
+minikube cluster
+
+The service can be started with the command:
+```bash
+minikube ingress resolve
+```
+
+This service will need to request privileged access from the user to be installed and started since updating of resolver
+configurations required administrative access on all known operating systems
+
+`resolve` is a sub-command of ingress because I think there is a strong use case for a different proposal which could be started with:
+
+```bash
+minikube ingress ssl
+```
+
+which will install local certificates on the host OS automatically for ingress resources. That is not covered in this
+proposal.
+
+#### 3. The background service can be automatically stopped and started when an addon is enabled or disabled
+This can reduce the steps required to enable this capability from 2 to 1. This is not covered as part of this proposal.
+
+## Alternatives Considered
+
+### Ways to access services
+* Don't use ingress at all and just use NodePort then match the DNS name using one of the alternatives described below
+  * Some services won't have a node port available. Especially if they are in a helm chart
+  * Now I have to remember the NodePort every time I want to access my service which maybe I can commit that 5 digit 
+    number to my memory for every service I need to access. If I don't have a good memory then this is really terrible
+    for me as I have to keep a reference sheet of where everything is running or I have to look it up with `kubectl`
+  * If the static ip address is somehow configured then I only have to create the DNS entries one time
+* Use `minikube tunnel` and an ingress dns resolver like CoreDNS then add a resolver file for the internal host IP 
```

Comment:
It's worth noting for this proposal that "tunnel" does update the host resolver on macOS.

We've been thinking more and more about background services, as we've been hitting limits with what is possible without one. This is especially true when we start to think about the minikube UI project. Here's some things we would also like to do with a daemon (outside of the scope of this proposal)

- Resource alerts ("kubectl get node" is noticing bad things)
- Time synchronization for hyperkit

Perhaps we should think about having a `minikubed` with this as one of the first features?

Reply:
`minikubed` sounds cool. How would we install the service? Like what would the command be?